app/src/org/commcare/tasks/ZipTask.java

```diff
@@ -235,133 +110,26 @@ private boolean zipFolder(File[] files, String zipFile, ZipOutputStream zos) thr
-                            } else {
-                                //Otherwise, the SD card just got removed, and we need to bail anyway.
-                                CommCareApplication._().reportNotificationMessage(NotificationMessageFactory.message(ProcessIssues.StorageRemoved), true);
-                                break;
-                            }
-                            continue;
-                        }
-
-                        //Check for success
-                        if (results[i].intValue() == FormUploadUtil.FULL_SUCCESS) {
-                            //FormRecordCleanupTask.wipeRecord(c, platform, record);
-                            //publishProgress(Localization.get("bulk.form.dialog.progress",new String[]{""+i, ""+results[i].intValue()}));
-                        }
-
-                        if (results[i].intValue() == FormUploadUtil.FAILURE) {
-                            publishProgress("Failure during zipping process");
-                            return null;
-                        }
-                    }
-                } catch (SessionUnavailableException sue) {
-                    this.cancel(false);
-                    return null;
-                } catch (Exception e) {
-                    //Just try to skip for now. Hopefully this doesn't wreck the model :/
-                    Logger.log(AndroidLogger.TYPE_ERROR_DESIGN, "Totally Unexpected Error during form submission" + getExceptionText(e));
-                }
-            }
-
-            long result = 0;
-            for (int i = 0; i < records.length; ++i) {
-                if (results[i] > result) {
-                    result = results[i];
-                }
-            }
-
-            if (result == 0) {
-                try {
-                    Log.d(TAG, "827 trying zip");
-
-                    String zipPath = CommCareWiFiDirectActivity.sourceDirectory;
-
-                    File nf = new File(zipPath);
-                    if (nf.exists()) {
-                        nf.delete();
-                    }
-
-                    zipTargetFolder(nf, CommCareWiFiDirectActivity.sourceZipDirectory);
-                    sourceDirectory.delete();
-                } catch (IOException ioe) {
-                    Log.d(TAG, "827 IOException: " + ioe.getMessage());
-                }
-            }
-            return records;
-        } else {
-            publishProgress("No forms to send.");
-            return null;
-        }
-    }
-
-    private String getExceptionText(Exception e) {
+    protected Integer doTaskBackground(Void... params) {
+        Log.d(TAG, "Doing UnzipTask");
         try {
-            ByteArrayOutputStream bos = new ByteArrayOutputStream();
-            e.printStackTrace(new PrintStream(bos));
-            return new String(bos.toByteArray());
-        } catch (Exception ex) {
-            return null;
+            String zipPath = toBeZippedPath;
+            File nf = new File(zipPath);

```

Comment:
There are 2 places in this method where you do `new File(toBeZippedPath)`, which is the only use of `toBeZippedPath`. I'd recommend doing it once in the constructor and storing it as a final class field instead of having `toBeZippedPath` be a class field.